cursor.go

```diff
@@ -0,0 +1,83 @@
+package gremgo
+
+import (
+	"context"
+	"io"
+	"sync"
+
+	"github.com/gedge/graphson"
+	"github.com/pkg/errors"
+)
+
+// Cursor allows for results to be iterated over as soon as available, rather than waiting for
+// a query to complete and all results to be returned in one block.
+type Cursor struct {
+	ID     string
+	mu     sync.RWMutex
+	eof    bool
+	buffer []string
+	client *Client
+}
+
+// Read a string response from the cursor, reading from the buffer of previously retrieved responses
+// when possible. When the buffer is empty, Read uses the cursor's client to retrieve further
+// responses from the database. As this function does not take context, a number of attempts
+// is hardcoded in refillBuffer() to prevent an infinite wait for further responses.
+func (c *Cursor) Read() (string, error) {
+	if len(c.buffer) == 0 {
+		if c.eof {
+			return "", io.EOF
+		}
+
+		if err := c.refillBuffer(); err != nil {
+			return "", err
+		}
+	}
+
+	s := c.buffer[0] + "\n"
```

Comment:
Consider moving the newline out into the observation StreamRowReader (https://github.com/ONSdigital/dp-graph/blob/develop/observation/reader.go#L60)

Reply:
Currently thinking this is unnecessary as the separation to Stream from Cursor more explicitly highlights that this is designed to return string rows